index.js

```diff
@@ -337,7 +337,7 @@
                 /* Optimal order of selectors for checking. */
                 var selector_elements = ['year', 'holiday', 'month', 'monthday', 'week', 'weekday'];
                 for (var selector_ind in selector_elements) {
-                    if (selectors[selector_elements[selector_ind]].length > 0) {
+                    if (selector_elements.hasOwnProperty(selector_ind) && selectors[selector_elements[selector_ind]].length > 0) {
```

Comment:
Ever since the [goto fail vulnerability](https://www.dwheeler.com/essays/apple-goto-fail.html) I try to use if with braces. Since the project does not have advanced unit testing and static code analysis yet, I would prefer that. Feel fee to add a linting tool to check for that. I have not really looked into eslint and others yet but I will.